lib/graphql/account_settings/cloud_object/address_cloud_object.dart

```diff
@@ -0,0 +1,99 @@
+import 'dart:convert';
+
+import 'package:autonomy_flutter/database/entity/wallet_address.dart';
+import 'package:autonomy_flutter/graphql/account_settings/account_settings_db.dart';
+
+class WalletAddressCloudObject {
+  final AccountSettingsDB _accountSettingsDB;
+
+  WalletAddressCloudObject(this._accountSettingsDB);
+
+  Future<void> deleteAddress(WalletAddress address) async {
+    await _accountSettingsDB.delete([address.key]);
+  }
+
+  Future<void> deleteAddressesByUuid(String uuid) async {
+    final addressesWithUUid = findByWalletID(uuid);
+    await _accountSettingsDB
+        .delete(addressesWithUUid.map((e) => e.key).toList());
+  }
+
+  List<WalletAddress> findAddressesWithHiddenStatus(bool isHidden) {
+    final allAddresses = getAllAddresses();
+    return allAddresses
+        .where((element) => element.isHidden == isHidden)
+        .toList();
+  }
+
+  WalletAddress? findByAddress(String address) {
+    final value = _accountSettingsDB.query([address]);
+    if (value.isEmpty) {
+      return null;
+    }
+    final addressJson = jsonDecode(value.first['value']!);
+    return WalletAddress.fromJson(addressJson);
+  }
+
+  List<WalletAddress> findByWalletID(String uuid) {
```

Comment:
what is the difference from `getAddressesByUuid `?

Reply:
just a copy from old method

Reply:
can we keep 1 only?

Reply:
i think i deleted it, it's marked 'Outdated'